File: android/src/main/java/com/github/quadflask/react/navermap/RNNaverMapPolylineOverlay.java
package com.github.quadflask.react.navermap;

import android.content.Context;

import com.naver.maps.geometry.LatLng;
import com.naver.maps.map.overlay.PolylineOverlay;

import java.util.List;

public class RNNaverMapPolylineOverlay extends ClickableRNNaverMapFeature<PolylineOverlay> {
    public RNNaverMapPolylineOverlay(EventEmittable emitter, Context context) {
        super(emitter, context);
        feature = new PolylineOverlay();
    }

    public void setCoords(List<LatLng> coords) {
        feature.setCoords(coords);
    }

    public void setLineWidth(float widthInScreenPx) {
        feature.setWidth(Math.round(widthInScreenPx));
    }

    public void setLineColor(int color) {
        feature.setColor(color);
    }

    public void setCapType(PolylineOverlay.LineCap value) {
        feature.setCapType(value);
    }

    public void setJoinType(PolylineOverlay.LineJoin value) {
        feature.setJoinType(value);
    }

    public void setPattern(int[] pattern) {
        feature.setPattern(pattern);
    }
}

File: android/src/main/java/com/github/quadflask/react/navermap/feature/RNNaverMapPolylineOverlay.kt
package com.github.quadflask.react.navermap.feature

import android.content.Context
import com.naver.maps.geometry.LatLng
import com.naver.maps.map.overlay.PolylineOverlay
import com.naver.maps.map.overlay.PolylineOverlay.LineCap
import com.naver.maps.map.overlay.PolylineOverlay.LineJoin
import kotlin.math.roundToInt

class RNNaverMapPolylineOverlay(
  emitter: EventEmittable,
  context: Context
) : ClickableRNNaverMapFeature<PolylineOverlay>(emitter, context) {
  init {
    feature = PolylineOverlay()
  }

  fun setCoords(coords: List<LatLng>) {
    feature.coords = coords
  }

  fun setLineWidth(widthInScreenPx: Float) {
    feature.width = widthInScreenPx.roundToInt()
  }

  fun setLineColor(color: Int) {
    feature.color = color
  }

  fun setCapType(value: LineCap?) {
    feature.setCapType(value)
  }

  fun setJoinType(value: LineJoin?) {
    feature.setJoinType(value)
  }

  fun setPattern(pattern: IntArray) {
    feature.setPattern(*pattern)
  }
}
